test/broadcasting.jl

```diff
@@ -223,309 +231,299 @@ end
     df = copy(refdf)
     df[:, 1] .+= [0, 1, 2] .+ 1
     @test df.x1 == [2.5, 4.5, 6.5]
-    @test df[2:end] == refdf[2:end]
+    @test df[:, 2:end] == refdf[:, 2:end]
 
     dfv = @view df[1:2, 2:end]
     dfv[:, 1] .+= [0, 1] .+ 1
     @test dfv.x2 == [5.5, 7.5]
-    @test dfv[2:end] == refdf[1:2, 3:end]
+    @test dfv[:, 2:end] == refdf[1:2, 3:end]
     @test Matrix(df) == [2.5  5.5  7.5  10.5  13.5
                          4.5  7.5  8.5  11.5  14.5
                          6.5  6.5  9.5  12.5  15.5]
 
     df = copy(refdf)
     dfv = @view df[1:2, 2:end]
     dfr = df[1, 3:end]
-    @test_throws DimensionMismatch df[1] .= rand(3, 1)
-    @test_throws DimensionMismatch dfv[1] .= rand(2, 1)
+    @test_throws DimensionMismatch df[!, 1] .= rand(3, 1)
+    @test_throws ArgumentError dfv[!, 1] .= rand(2, 1)
     @test_throws DimensionMismatch dfr[end-1:end] .= rand(3, 1)
     @test_throws DimensionMismatch df[:, 1] .= rand(3, 1)
     @test_throws DimensionMismatch dfv[:, 1] .= rand(2, 1)
-    @test_throws DimensionMismatch df[1] .= reshape(rand(3), :, 1)
-    @test_throws DimensionMismatch dfv[1] .= reshape(rand(2), :, 1)
+    @test_throws DimensionMismatch df[!, 1] .= reshape(rand(3), :, 1)
+    @test_throws ArgumentError dfv[!, 1] .= reshape(rand(2), :, 1)
     @test_throws DimensionMismatch dfr[end-1:end] .= reshape(rand(3), :, 1)
     @test_throws DimensionMismatch df[:, 1] .= reshape(rand(3), :, 1)
     @test_throws DimensionMismatch dfv[:, 1] .= reshape(rand(2), :, 1)
 
     df = copy(refdf)
-    df[:x1] .+= 1
+    df[!, :x1] .+= 1
     @test df.x1 == [2.5, 3.5, 4.5]
-    @test df[2:end] == refdf[2:end]
+    @test df[:, 2:end] == refdf[:, 2:end]
 
     dfv = @view df[1:2, 2:end]
-    dfv[:x2] .+= 1
-    @test dfv.x2 == [5.5, 6.5]
-    @test dfv[2:end] == refdf[1:2, 3:end]
-    @test Matrix(df) == [2.5  5.5  7.5  10.5  13.5
-                         3.5  6.5  8.5  11.5  14.5
-                         4.5  6.5  9.5  12.5  15.5]
+    @test_throws ArgumentError dfv[!, :x2] .+= 1
 
     dfr = df[1, 3:end]
     dfr[[:x4, :x5]] .= 10
     @test Vector(dfr) == [7.5, 10.0, 10.0]
-    @test Matrix(df) == [2.5  5.5  7.5  10.0  10.0
-                         3.5  6.5  8.5  11.5  14.5
+    @test Matrix(df) == [2.5  4.5  7.5  10.0  10.0
```

Comment:
??

Reply:
`dfv[!, :x2] .+= 1` above throws an error now in line 264 (`!` is disallowed in `SubDataFrames` as it indicates column replacement, while for `SubDataFrame` we can only do in-place assignment) therefore two entries in column `:x2` are not incremented by `1` in the test